frontend/src/TestEdit.js

```diff
@@ -0,0 +1,267 @@
+import React, { useEffect} from 'react';
+import  { useState } from 'react';
+import {Link, useParams} from 'react-router-dom';
+import {Button,  Container, Form, FormGroup, Input, Label} from 'reactstrap';
+import AppNavbar from './AppNavbar';
+import {useQuery} from "react-query";
+
+
+
+const emptyItem = {
+    test:{
+        name: '',
+        group:null,
+        author:null,
+        duration: '',
+        id: null
+    },
+
+    questions: [{
+        question:"",
+        id:1,
+        answerVariants:[""],
+        answers:[]
+
+    } ],
+
+    added:new Set(),
+    deleted:new Set(),
+    groupName:''
+
+};
+
+function TestEdit(){
+
+    function correctItem(data){
+        data.questions.forEach(function (q) {
+            if (q.id > maxId) {
+                maxId = q.id;
+            }
+        });
+
+        data.added = new Set();
+        data.deleted = new Set();
+        data.groupName = data.test.group.name;
+        setItem(data);
+
+    }
+
+
+    const [item, setItem] = useState(emptyItem);
+    var maxId=1;
+    const { id } = useParams();
+
+    const { error, isLoading} = useQuery('fullTests', () =>
+            fetch(`/subload/tests/${id}`).then(res => res.json()),
+        {
+            onSuccess: (data)=> correctItem(data),
+            enabled: id !== 'new'
+        }
+    );
+
+    if (error) return <div>Request Failed</div>;
+    if (isLoading) return <div>Loading...</div>;
+
+
+    function handleGroupChange(event) {
+        const target = event.target;
+        const value = target.value;
+        let newItem = {...item};
+
+
+        if(/(([a-z]{1,})|([A-Z]{1,}))-[1-9]{2}/.test(value)) {
```

Comment:
extract this regex as a const, move to another file and refer to it here

Reply:
Done